docs/command-reference.md

```diff
@@ -0,0 +1,709 @@
+# Container CLI Command Reference
```

Comment:
Do we need a PR template that includes a reminder (or checklist if we also want a reminder for unit/integration tests and anything else) for updating docs?

Reply:
if we wanted a more closed-form approach to this (one that we could validate via build), swift-argument-parser offers an `--experimental-dump-help` that writes all help output to a JSON file. If we wanted to merge informative content with the normative material from the dump, we could maintain the informative material as a set of files that could be JSON patched in, and then rendered to markdown or whatever we want